tests/unit/test_arg_recorder.py

```diff
@@ -136,16 +136,32 @@ def test_node_table(self):
         assert num_lines(nt) == self.arg.num_nodes + 1
         # one line for each node and one for header
 
-    def test_add_samples(self):
-        samples = [0, 1]
-        self.arg.add_samples(samples, 1.0)
+    def test_sample_table(self):
         out = six.StringIO()
         self.arg.dump_sample_table(out)
         print(out.getvalue())
-        assert num_lines(out.getvalue()) == len(samples) + 1  # for header
+        assert num_lines(out.getvalue()) == 1  # no samples
 
-    def test_sample_table(self):
+
+class TestARGrecorderGoodSampledPopulation():
+    def setup_method(self):
+        '''setup for all tests in class'''
+        self.arg = ARGrecorder()
+        self.arg.add_individual(2, time=3.0)
+        self.arg.add_individual(3, time=3.0)
+        self.arg.add_individual(4, time=1.0)
+        self.arg.add_record(left=0.5, right=1.0, parent=3, children=(4, ))
+        self.arg.add_record(left=0.0, right=0.5, parent=2, children=(4, ))
+        self.arg.add_individual(1, time=0.0)
+        self.arg.add_individual(0, time=0.0)
+        self.samples = [3, 4]
+        self.arg.add_samples(samples=self.samples, length=1.0)
```

Comment:
Oh, I see.  Your setup is in error - samples must have IDs that are 0..nsamples-1; but you've already added individuals 0 and 1, without specifying that they are samples, and so when add_samples tries to add these IDs, nothing happens.  If you'd left off lines 155 and 156, then add_samples would have added individuals 0 and 1, and properly flagged them as samples.

So, there's no samples, so the output nodetable of samples is empty.

To avoid this, we should put an 'else:' statement at
  https://github.com/ashander/ftprime/blob/master/ftprime/argrecorder.py#L37
that raises some kind of error saying the individual already exists.